### ISSUE TITLE
A better method for word embedding

### ISSUE THREAD

**Issue by kvmduc** [ansoncaonguyen/intro2ds-final-project#5]
I believed that the performance still can be improved since your method for word embedding is One hot encoding. 
One hot encoding just only check whether a word in lyric. So a large amount of information in lyric is wasted.  I think there are samples ,which has the same vector of lyric but have a different label

I recommend 2 other methods is `tf/idf` and `bag of words` which may improve your performance, you can put those module into pipeline and use directly

https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfVectorizer.html

https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html

How did you choose the hyperparameter `num_words = 15`. 
Choosing `num_words` larger or smaller may effect anything to your model ?

**Comment by ansoncaonguyen** [comment]
Thank you for your recommendations. 

We have updated the project using `bag of words` instead of `one hot encoding`. However, the performance of the models was not really improved. Personally, I think the reason behind the low performance of our models mainly depend on the size of our dataset.

We have also updated the effect when changing the hyperparameter `num_words` to our models.
